core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -493,6 +493,10 @@ public void validate() {
         throw new IllegalArgumentException("Value for Table Expiry period is valid only when file tracking table " +
                                              "is specified. Please specify file tracking table name.");
       }
+
+      if (!containsMacro("rowsLimit") && !StringUtils.isNumeric(rowsLimit)) {
+        throw new IllegalArgumentException("Invalid row limit. The value should be greater than zero.");
```

Comment:
incorrect message should be something like this
String.format("Invalid row limit, numeric value expected. value provided = {}", rowLimit)